runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -743,17 +743,20 @@ bitmapBaseType_t bitmapElemBit(size_t i) {
 
 static inline
 void bitmapClear(struct bitmap_t* b, size_t i) {
-  b->map[bitmapElemIdx(i)] &= ~bitmapElemBit(i);
+  if (b) {
+    b->map[bitmapElemIdx(i)] &= ~bitmapElemBit(i);
+  }
 }
 
 static inline
 void bitmapSet(struct bitmap_t* b, size_t i) {
+  assert(b);
   b->map[bitmapElemIdx(i)] |= bitmapElemBit(i);
 }
 
 static inline
 int bitmapTest(struct bitmap_t* b, size_t i) {
-  return (b->map[bitmapElemIdx(i)] & bitmapElemBit(i)) != 0;
+  return (b && (b->map[bitmapElemIdx(i)] & bitmapElemBit(i)) != 0);
```

Comment:
Its not totally clear to me why these changes where needed? When will `b` ever be `NULL`? Can all of these cases be `assert(b)` (like `bitmapSet`)?

Reply:
There is a lot of this idiom in the code:

```
    havePutsOut = (tcip->putVisBitmap != NULL
                   && bitmapTest(tcip->putVisBitmap, node));

```
I put the test for NULL into the function so I could clean iup my code:

```
  chpl_bool havePutsOut = bitmapTest(tcip->putVisBitmap, ab->node);
```

I plan to go back and clean up all the bitmap usage but will do that in a separate PR.

Reply:
I added the `assert` to `bitmapSet` because I'm paranoid.